### PR TITLE
Add open api doc generation task

### PR DESCRIPTION
**Pull request by loosebazooka**

### CODE REVIEW

File: src/main/java/com/google/cloud/tools/gradle/endpoints/framework/server/EndpointsServerPlugin.java [hunk 1]
```diff
@@ -118,6 +121,46 @@ public void execute(Project project) {
...
+  private void createGenerateOpenApiDocsTask() {
+    project
+        .getTasks()
+        .create(
+            GENERATE_OPENAPI_DOC_TASK,
+            GenerateOpenApiDocsTask.class,
+            new Action<GenerateOpenApiDocsTask>() {
+              @Override
+              public void execute(final GenerateOpenApiDocsTask genOpenApiDocs) {
+                genOpenApiDocs.setDescription("Generate endpoints openapi documents");
```
Comment (akerekes) [inline src/main/java/com/google/cloud/tools/gradle/endpoints/framework/server/EndpointsServerPlugin.java]: Is this user visible string? If so, I think it should use `Open API`.

File: src/main/java/com/google/cloud/tools/gradle/endpoints/framework/server/task/GenerateOpenApiDocsTask.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+  public void setHostname(String hostname) {
+    this.hostname = hostname;
+  }
+
+  /** Task entry point. */
+  @TaskAction
+  void generateOpenApiDocs() throws Exception {
+    getProject().delete(openApiDocDir);
+    openApiDocDir.mkdirs();
```
Comment (akerekes) [inline src/main/java/com/google/cloud/tools/gradle/endpoints/framework/server/task/GenerateOpenApiDocsTask.java]: Error handling?

File: src/main/java/com/google/cloud/tools/gradle/endpoints/framework/server/task/GenerateOpenApiDocsTask.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+    String classpath =
+        (getProject()
```
Comment (akerekes) [inline src/main/java/com/google/cloud/tools/gradle/endpoints/framework/server/task/GenerateOpenApiDocsTask.java]: Is the outer pair of parenthesis necessary?

File: src/test/java/com/google/cloud/tools/gradle/endpoints/framework/ProjectTests.java [hunk 4]
```diff
@@ -97,6 +98,12 @@ public void testServer() throws IOException, URISyntaxException {
     Assert.assertEquals(1, clientLib.getParentFile().listFiles().length);
     String apiJavaFile = getFileContentsInZip(clientLib, API_JAVA_FILE_PATH);
     Assert.assertThat(apiJavaFile, CoreMatchers.containsString(DEFAULT_URL_VARIABLE));
+
```
Comment (akerekes) [inline src/test/java/com/google/cloud/tools/gradle/endpoints/framework/ProjectTests.java:101]: I have a deja vu: weren't these tests separated into their own classes?

**Review by akerekes — APPROVED** [review]

File: src/test/java/com/google/cloud/tools/gradle/endpoints/framework/ProjectTests.java (hunk 4 above)
Comment (loosebazooka) [inline src/test/java/com/google/cloud/tools/gradle/endpoints/framework/ProjectTests.java:101]: Nah, this is gradle, that's the next step.

File: src/main/java/com/google/cloud/tools/gradle/endpoints/framework/server/task/GenerateOpenApiDocsTask.java (hunk 3 above)
Comment (loosebazooka) [reply src/main/java/com/google/cloud/tools/gradle/endpoints/framework/server/task/GenerateOpenApiDocsTask.java]: I can't remember if google java format adds it, or I did. But I'll try again.

**Pull request merged**